pkg/services/ngalert/api/api_prometheus.go

```diff
@@ -115,6 +122,19 @@ func (srv PrometheusSrv) RouteGetRuleStatuses(c *contextmodel.ReqContext) respon
 		return ErrResp(http.StatusBadRequest, errors.New("panel_id must be set with dashboard_uid"), "")
 	}
 
+	limitGroups, err := getInt64FromRequest(c.Req, "limit")
```

Comment:
```suggestion
	limitGroups, err := c.QueryInt64("limit")
```

Not sure if you know, but Grafana already has a set of methods that do this modulo the trimming of spaces (e.g. `c.QueryInt64()`. Is the trimming of spaces required? In the past, the technique I've used is to limit back to the default limit if we're unable to do the conversion - so, something like:

```
func (ctx *ReqContext) QueryInt64WithDefault(field string, d int64) int64 {
	n, err := strconv.ParseInt(ctx.Query(field), 10, 64)
	if err != nil {
		return d
	}
	return n
}
```

I'm not marking this as a nit, as I do think we want "pagination" by default, even though it's technically a breaking change. Perhaps we could have a special case to _not_ have pagination e.g. `-1`

Reply:
`c.QueryInt64()` is problematic for what we need to do here 🙁 It cannot differentiate between a valid value of `0`, an invalid value, and a missing value.